src/Microsoft.Data.Analysis/PrimitiveDataFrameColumnRollingWindow.cs

```diff
@@ -0,0 +1,183 @@
+                // This algorithm builds a sorted list. Time complexity is O(N) here since each value is added/removed to sortedList only once.
+                // If the values at the end of the sortedList are lesser than value, delete them since they can never be the max in this window
+                // If value is null, treat it as double.MinValue
+                while (sortedList.Count != 0 && sortedList.Last.Value.Key <= doubleValue)
+                {
+                    sortedList.RemoveLast();
+                    windowValues.RemoveLast();
+                }
+
+                sortedList.AddLast(new KeyValuePair<double, long>(doubleValue, currentIndex));
+                windowValues.AddLast(value);
+
+                while (sortedList.First.Value.Value <= currentIndex - _windowSize)
+                {
+                    sortedList.RemoveFirst();
+                    windowValues.RemoveFirst();
+                }
+
+                // For the edge case of (6, null, null, null) where (null, null, null) is in the current window, we will have (null) in sortedList at this point. Check if value is null and sortedList.First is double.Max. If it is, remove it
+                if (!value.HasValue && sortedList.Count == 1 && sortedList.Last.Value.Key == double.MinValue)
+                {
+                    Debug.Assert(windowValues.Count == 1);
+                    sortedList.RemoveFirst();
+                    windowValues.RemoveFirst();
+                }
+
+                currentIndex++;
+                if (currentIndex < _windowSize)
+                {
+                    return null;
+                }
+
+                return windowValues.Count > 0 ? windowValues.First.Value : null;
+            });
+
+        }
+
+        public override PrimitiveDataFrameColumn<int> Count()
+        {
+            long currentIndex = 0;
+            int count = 0;
+            LinkedList<T?> windowValues = new LinkedList<T?>();
+            return _currentColumn.Apply<int>((T? value) =>
+            {
+                windowValues.AddLast(value);
+                if (currentIndex >= _windowSize)
+                {
+                    if (windowValues.First.Value.HasValue && count > 0)
+                    {
+                        count--;
+                    }
+                    windowValues.RemoveFirst();
+                }
+                if (value.HasValue)
+                {
+                    count++;
+                }
+                if (currentIndex < _windowSize - 1)
+                {
+                    currentIndex++;
+                    return null;
+                }
+                currentIndex++;
+                return count;
+            });
+        }
+    }
+
+    public partial class PrimitiveDataFrameColumn<T> : DataFrameColumn
+        where T : unmanaged
+    {
+        internal PrimitiveDataFrameColumn<U> ApplyRollingFunc<U>(Func<LinkedList<T?>, long, U?> func, int windowSize)
```

Comment:
Why not put this in the file that contains the rest of the class? It is a bit confusing when a class is spread all over the place - and with other classes in the same file.